contracts/tokenomics/generator/src/contract.rs

```diff
@@ -614,50 +558,64 @@ fn set_tokens_per_block(
 }
 
 /// # Description
-/// Updates the amount of accrued rewards for all generators. Returns a [`ContractError`] on failure, otherwise
-/// returns a [`Response`] with the specified attributes if the operation was successful.
+/// Updates reward variables for all pools. Returns an [`ContractError`] on failure, otherwise
+/// returns the [`Response`] with the specified attributes if the operation was successful.
 /// # Params
-/// * **deps** is an object of type [`DepsMut`].
+/// * **deps** is the object of type [`DepsMut`].
 ///
-/// * **env** is an object of type [`Env`].
+/// * **env** is the object of type [`Env`].
 pub fn mass_update_pools(mut deps: DepsMut, env: Env) -> Result<Response, ContractError> {
     let response = Response::default();
 
     let cfg = CONFIG.load(deps.storage)?;
-    let pools = get_pools(deps.storage);
 
-    if pools.is_empty() {
+    if cfg.active_pools.is_empty() {
         return Ok(response);
     }
 
-    for (lp_token, mut pool) in pools {
-        accumulate_rewards_per_share(deps.branch(), &env, &lp_token, &mut pool, &cfg, None)?;
-        POOL_INFO.save(deps.storage, &lp_token, &pool)?;
+    for (lp_token, _) in &cfg.active_pools {
+        let mut pool = POOL_INFO.load(deps.storage, lp_token)?;
+
+        accumulate_rewards_per_share(deps.branch(), &env, lp_token, &mut pool, &cfg, None)?;
+        POOL_INFO.save(deps.storage, lp_token, &pool)?;
     }
 
     Ok(response.add_attribute("action", "mass_update_pools"))
 }
 
 /// # Description
-/// Updates the amount of accrued rewards for a specific generator. Returns a [`ContractError`] on
+/// Updates rewards and returns it to user. Returns a [`ContractError`] on
 /// failure, otherwise returns a [`Response`] with the specified attributes if the operation was successful.
 /// # Params
 /// * **deps** is an object of type [`DepsMut`].
 ///
 /// * **env** is an object of type [`Env`].
 ///
-/// * **lp_token** is an object of type [`Addr`]. This is the LP token whose accrued rewards we update.
-pub fn update_pool(mut deps: DepsMut, env: Env, lp_token: Addr) -> Result<Response, ContractError> {
+/// * **lp_token** is the object of type [`Addr`]. Sets the liquidity pool to be updated.
+///
+/// * **account** is the object of type [`Addr`].
+pub fn claim_rewards(
+    mut deps: DepsMut,
+    env: Env,
+    lp_token: Addr,
```

Comment:
I think we might want to claim rewards by several lp tokens and not only one. Maybe let's use vector instead? Then we can use mass_update_pools inside

Reply:
Done, thanks